src/main/java/org/dependencytrack/persistence/ComponentQueryManager.java

```diff
@@ -241,6 +241,7 @@ public PaginatedResult getComponents(final Project project, final boolean includ
         if (filter != null) {
             queryString += """
                     AND LOWER("A0"."NAME") LIKE ('%' || LOWER(:nameFilter) || '%')
+                    OR LOWER("A0"."GROUP") LIKE ('%' || LOWER(:nameFilter) || '%')
```

Comment:
The filters should be enclosed in parentheses:

```suggestion
                    AND (LOWER("A0"."NAME") LIKE ('%' || LOWER(:nameFilter) || '%')
                    OR LOWER("A0"."GROUP") LIKE ('%' || LOWER(:nameFilter) || '%'))
```

`select false and false or true` evaluates to `true`. 

In this specific case, the condition would match if only the component's group matches the query, even if it belongs to a different project (`"A0"."PROJECT_ID" = :projectId` is `false`).

Adding parens like this: `select false and (false or true)` would evaluate to `false` as expected. The original PR had these parentheses.